src/components/OptionListContextProvider.tsx

```diff
@@ -0,0 +1,137 @@
+import React, {createContext, useCallback, useContext, useEffect, useMemo, useRef, useState} from 'react';
+import {withOnyx} from 'react-native-onyx';
+import type {OnyxCollection} from 'react-native-onyx';
+import * as OptionsListUtils from '@libs/OptionsListUtils';
+import type {OptionList} from '@libs/OptionsListUtils';
+import * as ReportUtils from '@libs/ReportUtils';
+import ONYXKEYS from '@src/ONYXKEYS';
+import type {Report} from '@src/types/onyx';
+import {usePersonalDetails} from './OnyxProvider';
+
+type OptionsListContextProps = {
+    /** List of options for reports and personal details */
+    options: OptionList;
+    /** Function to initialize the options */
+    initializeOptions: () => void;
+    /** Flag to check if the options are initialized */
+    areOptionsInitialized: boolean;
+};
+
+type OptionsListProviderOnyxProps = {
+    /** Collection of reports */
+    reports: OnyxCollection<Report>;
+};
+
+type OptionsListProviderProps = OptionsListProviderOnyxProps & {
+    /** Actual content wrapped by this component */
+    children: React.ReactNode;
+};
+
+const OptionsListContext = createContext<OptionsListContextProps>({
+    options: {
+        reports: [],
+        personalDetails: [],
+    },
+    initializeOptions: () => {},
+    areOptionsInitialized: false,
+});
+
+function OptionsListContextProvider({reports, children}: OptionsListProviderProps) {
+    const areOptionsInitialized = useRef(false);
+    const [options, setOptions] = useState<OptionList>({
+        reports: [],
+        personalDetails: [],
+    });
+    const personalDetails = usePersonalDetails();
+
+    useEffect(() => {
+        // there is no need to update the options if the options are not initialized
+        if (!areOptionsInitialized.current) {
+            return;
+        }
+
+        const lastUpdatedReport = ReportUtils.getLastUpdatedReport();
+        const newOption = OptionsListUtils.createOptionFromReport(lastUpdatedReport, personalDetails);
+        const replaceIndex = options.reports.findIndex((option) => option.reportID === lastUpdatedReport.reportID);
+
+        if (replaceIndex === undefined) {
```

Comment:
isn't that this won't be never passed because `findIndex` will return index of the first element or -1.